crates/core/component/sct/src/component/rpc.rs

```diff
@@ -35,4 +38,21 @@ impl QueryService for Server {
             epoch: Some(epoch.into()),
         }))
     }
+
+    #[instrument(skip(self, request))]
+    async fn anchor_by_height(
+        &self,
+        request: tonic::Request<AnchorByHeightRequest>,
+    ) -> Result<tonic::Response<AnchorByHeightResponse>, Status> {
+        let state = self.storage.latest_snapshot();
+
+        let anchor = state
+            .get_anchor_by_height(request.get_ref().height)
+            .await
+            .map_err(|e| tonic::Status::unknown(format!("could not get anchor for height: {e}")))?;
```

Comment:
minor point but this will display the error message, not the height, good idea to include both

Reply:
agreed. so something like this?
```rust
let height = request.get_ref().height;
let anchor = state
    .get_anchor_by_height(height)
    .await
    .map_err(|e| { tonic::Status::unknown(format!("could not get anchor for height {height}: {e}"))})?;
```

Reply:
that would be great 🙏

Reply:
do you prefer --fixup commits or should I add the changes separately?

Reply:
I'll squash so whatever works for you, adding the changes separately sounds good